docs/Theory/Math/lagrange polynomial.md

```diff
@@ -1 +1,51 @@
 # Lagrange Polynomial
+
+Polynomial interpolation is the interpolation of a given data set by the polynomial of lowest possible degree that passes through the points of the dataset.
+Lagrange interpolating polynomial is the unique polynomial of lowest degree that interpolates a given set of data.
+
+## Definition
+
+The Lagrange interpolating polynomial for nodes ${y_0, y_1, ..., y_k}$ is the linear combination:
+
+$$\sum_{j=0}^{k} {y_j*l_j(x)}$$
+
+Where $l_j(x)$ is Langrange basis (polynomial): 
+ 
+$$
+\mathcal{l_j}(x) = \prod_{m \neq j, m >= 0}\frac{x -x_m}{x_j - x_m}
+$$
+
+As mentioned above the interpolating polynomial is unique. This characteristic allows uses in cryptography such as: Shamir's secret sharing scheme and KZC polynomial commitmens (Kate commitment) *[https://en.m.wikipedia.org/wiki/Lagrange_polynomial]*
```

Comment:
The commitment scheme is referred to as KZG.

Typo: commitmens -> commitments

Can you add a reference for KZG?

Reply:
I added the original reference for KZG:

https://www.iacr.org/archive/asiacrypt2010/6477178/6477178.pdf